engine/src/main/java/org/hibernate/validator/internal/engine/ValidationContext.java

```diff
@@ -68,6 +76,12 @@
 	private final Class<T> rootBeanClass;
 
 	/**
+	 * The metadata of the root bean.
+	 */
+	@Immutable
```

Comment:
Yeah right, I used it as a marker to say that this value was not supposed to change during the object lifecycle and it was supposed to be "final" as the other ones. I'll remove them.